### PR TITLE
 piControl: increase priority of io-thread

### PR DESCRIPTION
**Pull request by linosanfilippo-kunbus**
During the piBridge communication lots of messages are logged in the kernel log that indicate UART data reception errors like parity, frame and (occassionally) overflow errors.

As it turned out increasing the realtime priority of the thread that handles the piBridge communication to 54 (which is the same as the SPI thread priority) fixes these issues.

Before this change an error rate (ratio of total errors to received bytes) of 0,000812359 (i.e. 0,081%) could be observerd during a 45 minute test. After this change the error rate is decreased to 0 (i.e. no errors at all). This rate is even below the error rate that is observed with the 4.19 Revpi kernel (0,0000262%)

NOTE: The reason for this effect is still not completely clear, presumably the UART has to switch as quickly as possible from transmitting to receiving data.

### CODE REVIEW

File: project.h [hunk 1]
```diff
@@ -92,7 +92,14 @@
 #define DEBUG_DEVICE_IO
 
 #define RT_PRIO_UART    39
-#define RT_PRIO_BRIDGE  38
+/*
+ * The priority has to be at least 54 (same as SPI thread), otherwise lots
+ * of UART reception errors are observed. The reason for this effect is not
+ * yet completely clear, presumably the UART has to switch as quick as
```
Comment (l1k) [inline project.h:98]: quick*ly*

File: project.h [hunk 2]
```diff
@@ -92,7 +92,14 @@
 #define DEBUG_DEVICE_IO
 
 #define RT_PRIO_UART    39
```
Comment (l1k) [inline project.h:94]: It seems `RT_PRIO_UART` is obsolete on the serdev branch.

File: project.h [hunk 3]
```diff
@@ -92,7 +92,14 @@
...
-#define RT_PRIO_BRIDGE  38
+/*
+ * The priority has to be at least 54 (same as SPI thread), otherwise lots
+ * of UART reception errors are observed. The reason for this effect is not
+ * yet completely clear, presumably the UART has to switch as quick as
+ * possible from data transmission to data reception to avoid errors during
+ * piBridge communication.
+*/
+#define RT_PRIO_BRIDGE  54
 #define RT_PRIO_GATE    37
```
Comment (l1k) [inline project.h:103]: `RT_PRIO_GATE` is obsolete since 2de8667a34, should have been removed by that commit. :(

**Comment by l1k** [comment]
Thanks for bottoming out on this, good catch. Turns out the priority *was* sufficiently high until 0ae981ad6 reduced it 5 years ago. :(

File: project.h (hunk 2 above)
Comment (linosanfilippo-kunbus) [reply project.h:94]: Hmm, this (and RT_PRIO_GATE) is indeed obsolete but is not part of the fix. Should it be in this patch nevertheless?

**Comment by nbuchwitz** [comment]
Should we maybe remove the unused ttyAMA constant in e3bdc02 too?

**Comment by l1k** [comment]
Could you s/quick/quickly/ in that code comment as requested please? (If only to assuage my grammar OCD... ;) )

**Review by l1k — CHANGES_REQUESTED** [review]
_

File: project.h (hunk 1 above)
Comment (linosanfilippo-kunbus) [reply project.h]: ah, sorry I just realized that you referred to _another_ typo in the comment section of the source code (I already fixed the same typo in the commit message)

**Review by l1k — APPROVED** [review]

**Pull request merged**